### PR TITLE
[cloud](storage vault) Rewrite vaults regression test and fix two error

### PR DESCRIPTION
**Pull request by swjtu-zhanglei**
* Rewrite vaults regression test
* Fix show storage vault stmt `default` column incorrectly
* Fix create table cannot find storage vault after creating the vault

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by SWJTU-ZhangLei** [comment]
run buildall

File: regression-test/suites/vault_p0/default/test_default_vault.groovy [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        sql """
+            CREATE TABLE ${tableName} (
+                `key` INT,
+                value INT
+            ) DUPLICATE KEY (`key`) DISTRIBUTED BY HASH (`key`) BUCKETS 1
+            PROPERTIES ('replication_num' = '1')
+        """
+
+        sql """ insert into ${tableName} values(1, 1); """
+        qt_sql """ select * from ${tableName}; """
```
Comment (gavinchou) [inline regression-test/suites/vault_p0/default/test_default_vault.groovy:95]: just check the result of the select stmt in memory 
we dont need to  manage another file called test_default_vault.out
```
result =  sql """ select * ..."""
assertEqual(result.size(), 1);
assertEqual(result[0][0], 1);
...

**Review by gavinchou — DISMISSED** [review]

File: regression-test/suites/vault_p0/default/test_default_vault.groovy (hunk 1 above)
Comment (SWJTU-ZhangLei) [reply regression-test/suites/vault_p0/default/test_default_vault.groovy]: > just check the result of the select stmt in memory we dont need to manage another file called test_default_vault.out
> 
> ```
> result =  sql """ select * ..."""
> assertEqual(result.size(), 1);
> assertEqual(result[0][0], 1);
> ...
> ```

done

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**